samza-core/src/main/java/org/apache/samza/storage/TaskSideInputStorageManager.java

```diff
@@ -74,7 +74,7 @@
   private final SystemAdmins systemAdmins;
   private final TaskName taskName;
   private final TaskMode taskMode;
-  private final Map<SystemStreamPartition, String> lastProcessedOffsets = new ConcurrentHashMap<>();
```

Comment:
Perhaps undo this change. 
Since this map actually stores the "last processed offset for the given side input" (which is used to emit a metric). 
SideInputs are only checkpointed locally (see writeOffsetFiles).

Reply:
I was trying to clarify that this field no longer stored the actual "last processed offsets", since we now need to store the "checkpoint offset" for writing to the offset files.
Do you think we should also track the actual "last processed offsets" explicitly for the metric? Or would the "checkpoint offset" be sufficient for the metric?